### PR TITLE
Ensure Site object in global scope is NilSite and never persisted.

### LINKED ISSUE #1041: First User created in global scope creates an empty global Site object and "admin" Role
### Descriptive summary

While this is not a severe issue, it is easily reproducible.  When you run Hyku in a multitenant environment, the first global user you create generates a global "admin" Role and an empty global Site object.

### Expected behavior

The global "admin" Role and Site object should never exist, as they have no meaning in the global scope (i.e. outside of a specific tenant).

### Steps to reproduce the behavior

1. Start up Hyku with multitenancy enabled (e.g. https://github.com/projecthydra-labs/hyku/wiki). Make sure your database is initially empty (e.g. `rails db:drop db:create db:migrate`)
2. In a second window, start up a rails console (`rails c`). `User.all`, `Site.all` and `Role.all` should all be empty initially.
3. From the "Get Started" splash screen, create a new global user. Do this by clicking on "Administrator login" in the footer, and signup for a new user account.
4. Go back to your Rails console, and type those three commands again:
    * `User.all` will show your newly created user. That's correct.
    * `Site.all` will now show an empty Site object. That shouldn't exist
    * `Role.all` will now show a new "admin" Role relating to that empty Site object. If you type `User.first.site_roles` you'll see that your user is now an "admin" on that empty Site.

This empty `Site` and global "admin" `Role` do not seem harmful, but they are very confusing when you are using Rails console or looking at the data in the DB. They have no connection to a specific tenant, and no purpose outside of a tenant.

### Related work

This seems to relate to the fact that the "admin" role is being created automatically for the first user account, even if that User is a global user.  The "admin" role should likely only be added for the first user account on a *tenant*.  

Here's the suspect line of code, which creates that "admin" Role and the new empty Site: https://github.com/projecthydra-labs/hyku/blob/master/app/models/user.rb#L55

This issue also seems like it should be resolved by PR #154, if/when that work is completed. That PR seems to include better logic to ensure the "admin" Role is only generated on a tenant (and not in the global scope).

### PR DESCRIPTION
**Pull request by tdonohue**
Fixes #1041

While this isn't a serious issue, it's extremely annoying/confusing to debugging other issues (see #1041). So, I'm scratching my own itch and simply fixing it.

This PR creates a new `NilSite` object which is used to represent the `Site` object in a global scope (i.e. global tenant in a multitenant environment). This `NilSite` ensures that an empty Site is not persisted to the database (and admin rights assigned) as described in #1041.

This PR comes with specs, but can also be tested by repeating the behavior described in #1041 and ensuring that an empty Site is not generated in the global / default database schema.

Comments/suggestions welcome, but this was the easiest way I could figure out to resolve this strange behavior.

@projecthydra-labs/hyrax-code-reviewers

### CODE REVIEW

**Comment by mjgiarlo** [comment]
@atz Would you mind giving this a review?

File: app/models/account.rb [hunk 1]
```diff
@@ -63,6 +63,14 @@ def self.single_tenant_default
     end
   end
 
+  # @return [Boolean] whether this Account is the global tenant in a multitenant environment
+  def self.global_tenant?
+    # Global tenant only exists when multitenancy is enabled and NOT in test environment
+    # (Test environment always acts like a single tenant, i.e. tenant switching not possible)
```
Comment (atz) [inline app/models/account.rb]: This is a partially misleading statement about test behavior.  See [multitenancy_metadata.rb#L12-L22](https://github.com/samvera-labs/hyku/blob/master/spec/support/multitenancy_metadata.rb#L12-L22) for more detail.  In short, there are many ways tests do **not** behave like single tenant: the most common default is `:multitenant`, and `:singletenant` is the exception!

File: spec/models/nil_site_spec.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
+require 'spec_helper'
```
Comment (atz) [inline spec/models/nil_site_spec.rb]: This line is unnecessary.

File: app/models/nil_site.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+class NilSite
+  class << self
+    # NilSite must be a singleton like Site
+    attr_writer :instance
+    def instance
+      @instance ||= NilSite.new
+    end
+  end
+
+  def id
```
Comment (atz) [inline app/models/nil_site.rb]: You can replace all this `nil`-returning methods with a single `attr_reader` declaration.

File: app/models/account.rb (hunk 1 above)
Comment (tdonohue) [inline app/models/account.rb]: Thanks @atz. Aha, yes, I didn't mean to state/imply `:singletenant` is the default (as I realize it isn't). The important point in the comment is the part at the end: `tenant switching is currently not possible`, and that is the reason why a global_tenant doesn't make any sense in the test environment (i.e. `global_tenant?==false`). 

Would it be reasonable to replace this last line with simply saying: 
`(In test environment tenant switching is currently not possible)`

Comment (atz) [inline app/models/account.rb]: Yeah, something like that would be clearer.

**Comment by tdonohue** [comment]
After a long hiatus, I finally got back to finishing up this PR.

@atz : I've made all the (minor) changes you've suggested. Thanks for the feedback in improving this!

This is ready for a re-review from someone... maybe @atz or @mjgiarlo?

**Comment by mjgiarlo** [comment]
@atz If you're available, could you re-review? If not, would you mind dismissing your review? Thank you, sir. I believe @tdonohue has responded to all of your comments.
